library/core/src/ptr/non_null.rs

```diff
@@ -594,7 +594,7 @@ impl<T: ?Sized> NonNull<T> {
     #[rustc_const_stable(feature = "non_null_convenience", since = "1.80.0")]
     #[requires(
         (self.as_ptr().addr() as isize).checked_add(count).is_some() &&
-        kani::mem::same_allocation(self.as_ptr(), self.as_ptr().wrapping_byte_offset(count))
+        ub_checks::same_allocation(self.as_ptr(), self.as_ptr().wrapping_byte_offset(count))
```

Comment:
Nit pick: earlier we had `core::ub_checks::same_allocation`, here it's just `ub_checks::same_allocation`. I'd expect those to be consistent?

And if you agree: we should review all such cases to establish consistency.